flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/DryRun.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2010-2016 Boxfuse GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.flywaydb.core.internal.dbsupport;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Savepoint;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.WeakHashMap;
+import org.flywaydb.core.api.FlywayException;
+
+public class DryRun {
+    private static final String SAVEPOINT_MARKER = "dryrun";
+    private final Map<Connection, Savepoint> dryRunSavepoints = new WeakHashMap<Connection, Savepoint>();
+    private final int connectionCount;
+    
+    public DryRun(int transactionIsolationLevel, DbSupport dbSupport, Connection... connections) {
+        if (dbSupport == null) {
+            throw new NullPointerException("A Dry-run requires dbSupport to determine whether connections can support a rollback successfully.");
+        }
+
+        if (!dbSupport.supportsDdlTransactions()) {
```

Comment:
that is why there are test and prelive environments. going straight from dev to production is cowboy programming. At the moment you are trying to implement feature as a workaround for flaws in your development process (not having proper test and prelive environments to safely test releases). 

but not everything is bad - I would use dry run to test my migrations only in development. It may be good/faster way to flush out syntax errors that I occasionally make.